src/legacy/core_plugins/kibana/public/visualize/np_ready/editor/visualization_editor.js

```diff
@@ -42,8 +55,7 @@ export function initVisEditorDirective(app, deps) {
             timeRange: $scope.timeRange,
             filters: $scope.filters,
             query: $scope.query,
-            appState: $scope.appState,
-            linked: !!$scope.savedObj.savedSearchId,
```

Comment:
if you save this prop passing down into react component, it'll be much more easier to achieve the correct behavior with linked search.